### PR TITLE
pagination search, fix getCats infinite loop

### PR DESCRIPTION
**Pull request by GrayDust404**
I'm managing up to 17000 archives with lanraragi. lanraragi-react-web is loading very slowly when I switch to search page because it fetch all archive info at one time. To make a better eexperience, I try to modify some code to make it query archived as needed.

I find when I don't have any categories on lanrargi, lanraragi-react-web will call getCats again and again. This PR also fixed it

### CODE REVIEW

**Comment by hibikikuze4dan** [comment]
Hey, thanks for contributing! The fetching all archives behavior was intentional. It was something I wanted that all of the other apps that existed at the time were either not doing or not doing to my satisfaction. I'm fine with adding and making use of the api's pagination, but I'd rather it be a setting to choose than default behavior. If you can make that happen then I'd be glad to merge it in 👍 .

Good find with the categories issue 👍 !

Also, sorry in advance if my responses are slow on this. Currently on vacation 🌴

**Comment by GrayDust404** [comment]
Thanks for your suggestion! Now the settings page has a switch to control either use server-side pagination or local pagination.

File: src/app/slice.js [hunk 1]
```diff
@@ -17,7 +18,10 @@ const initialState = {
   pages: [],
   randomArchives: [],
   renderedPages: [],
-  searchArchives: [],
+  searchArchives: {
```
Comment (hibikikuze4dan) [inline src/app/slice.js]: Turning this into an object breaks one of the slice updating functions. Let's keep this as an array and if you really need to save the total to the state then just add it to the slice as its own value like `totalArchives` or something along those lines.

**Comment by hibikikuze4dan** [comment]
There are a lot of lint warnings in the code added, could you please clean them up where possible.

**Comment by GrayDust404** [comment]
1. Empty the searchArchives and re-query if user changed pagination setting
2. Cleaned lint warns in the code added

**Comment by GrayDust404** [comment]
I found that lanraragi does not control the size of the returned data according to the length field, and more modifications are needed.😢

**Review by hibikikuze4dan — APPROVED** [review]
Looking good! I don't see any more issues 👍 . Will merge.

**Pull request merged**